.travis.yml

```diff
@@ -0,0 +1,29 @@
+# We will use docker to set up out environment, so don't use any particular
+# language in Travis itself
+language: generic
+
+# Enable docker
+sudo: required
+services:
+    - docker
+
+env:
+    global:
+        - LATEST_DIST=xenial
+        - BRANCH="$TRAVIS_BRANCH"
+        - TAG="$TRAVIS_TAG"
+    matrix:
+        - DIST=trusty
+        - DIST=xenial
+
+script: make -Brj2 build
+
+deploy:
+    provider: script
+    script:
+        set -eux ;
+        docker login -u "$DOCKER_USER" -p "$DOCKER_PASS" ;
+        make -rj2 push
```

Comment:
Travis doesn't support multiple commands for the `deploy:` `script:`. And it interprets the `\n` literally when using the *heredoc* format of yml (`script: |`), so the only way I found to make it work for a multi-line script was like this (I didn't want to move to a separate file so one can see the relation between the main script and the deploy script more easily).

Maybe this deserves a comment.